### PR TITLE
feat(integrations): Better user messaging on install-from-provider flow

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Adds feature list when organization is select, with feature gate upgrade buttons (for sentry.io)

 - Adds warning message if the user does not have permissions to install the integration to the selected org.

test
tunnel evanpurkhiser com_extensions_external-install_363188_ 1
tunnel evanpurkhiser com_extensions_external-install_363188_ 2

### CODE REVIEW

File: src/sentry/static/sentry/app/views/integrationInstallation.jsx [hunk 1]
```diff
@@ -1,112 +1,151 @@
...
-    window.addEventListener('message', this.receiveMessage, false);
-  }
+import SelectControl from 'app/components/forms/selectControl';
 
-  componentWillUnmount() {
-    window.removeEventListener('message', this.receiveMessage);
+/**
+ * Currently the only integration that has a install-from-provider flow is
+ * Github, so we hardcode this to ONLY support he github provider at the
+ * moment. We'll come back later and handle more later.
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/integrationInstallation.jsx]: What about jira cloud?

**Comment by mnoble** [comment]
What happens when they click "Upgrade"? Does it break the installation process?

**Comment by EvanPurkhiser** [comment]
@mnoble Where would the click upgrade?

File: src/sentry/static/sentry/app/views/integrationInstallation.jsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/integrationInstallation.jsx]: Jira cloud doesn't push you through this flow

**Comment by mnoble** [comment]
@EvanPurkhiser in the "Small billing plans and above" component. Presumably that takes you to the billing page?

**Comment by EvanPurkhiser** [comment]
Oh I see, sorry, I was thinking "Upgrade" old plugins. Yeah it takes you to the billing page, Unfortunately there wouldn't be a completely unified flow here.

**Comment by benvinegar** [comment]
Can we disable/grey out the upgrade button when you come through this flow. Or just remove it, honestly.

**Comment by EvanPurkhiser** [comment]
> Can we disable/grey out the upgrade button when you come through this flow. Or just remove it, honestly.

Decided offline to not do this.

File: src/sentry/static/sentry/app/views/integrationInstallation.jsx [hunk 2]
```diff
@@ -1,112 +1,169 @@
...
+  state = {
+    selectedOrg: null,
+    organization: null,
+    providers: [],
+    reloading: false,
+  };
 
   getEndpoints() {
-    return [['organizations', '/organizations/?owner=1']];
+    return [['organizations', '/organizations/']];
```
Comment (mnoble) [inline src/sentry/static/sentry/app/views/integrationInstallation.jsx]: Won't this receive all Orgs, regardless of ownership/membership? ie. Orgs the User has no access to.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/integrationInstallation.jsx]: What do you mean all orgs? Like as in _all_ orgs entirely? 

https://github.com/getsentry/sentry/blob/230f91904e4b596f652e62333a0b013866773bf2/src/sentry/api/endpoints/organization_index.py#L61-L64

Comment (mnoble) [reply src/sentry/static/sentry/app/views/integrationInstallation.jsx]: Where does it actually apply that logic, though? I'm probably missing something, but all I see it doing in this case is `Organization.objects.distinct()` and then serializing that.

**Review by mnoble — APPROVED** [review]

**Pull request merged**